tests/test_mixins.py

```diff
@@ -23,6 +30,64 @@ def setup(self):
         AnotherSingletonDerived._instances = {}
         SingletonDerived._instances = {}
 
+    def test_get_instance_key_without_arguments(self):
+        instance_key = SingletonClass._get_instance_key()
+        assert instance_key == 'tests.test_mixins.SingletonClass||'
+
+    @pytest.mark.parametrize(
+        'args,instance_key',
+        [
+            (
+                ('sum', 67, 22.12),
+                'tests.test_mixins.SingletonClass|sum-67-22.12|',
+            ),
+            (
+                ('person', {'name': 'John'}),
+                'tests.test_mixins.SingletonClass|person-{\'name\': \'John\'}|',  # noqa
+            ),
+        ],
+    )
+    def test_get_instance_key_with_args(self, args, instance_key):
+        assert SingletonClass._get_instance_key(*args) == instance_key
+
+    @pytest.mark.parametrize(
+        'kwargs,instance_key',
+        [
+            (
+                {'calc': 'mult', 'a': 23, 'b': 10.12},
+                'tests.test_mixins.SingletonClass||a:23-b:10.12-calc:mult',
+            ),
+            (
+                {'product': {'sku': '123'}},
+                'tests.test_mixins.SingletonClass||product:{\'sku\': \'123\'}',
+            ),
+        ],
+    )
+    def test_get_instance_key_with_kwargs(self, kwargs, instance_key):
+        assert SingletonClass._get_instance_key(**kwargs) == instance_key
+
+    @pytest.mark.parametrize(
+        'args,kwargs,instance_key',
+        [
+            (
+                ('sum', 67, 22.12),
+                {'calc': 'mult', 'a': 23, 'b': 10.12},
+                'tests.test_mixins.SingletonClass|sum-67-22.12|a:23-b:10.12-calc:mult',  # noqa
+            ),
+            (
+                ('person', {'name': 'John'}),
+                {'product': {'sku': '123'}},
+                'tests.test_mixins.SingletonClass|person-{\'name\': \'John\'}|product:{\'sku\': \'123\'}',  # noqa
+            ),
+        ],
+    )
+    def test_get_instance_key_with_args_and_kwargs(
+        self, args, kwargs, instance_key
+    ):
+        assert (
+            SingletonClass._get_instance_key(*args, **kwargs) == instance_key
+        )
+
```

Comment:
I think the `instance_key` generation should not be tested. From the perspective of someone using the Mixin, the important thing is the guarantee of the Singleton behavior when the parameters are provided, how it does that is not important and the tests should not fail if we find a better way to do that internally.

Reply:
Fixed :+1: